src/main/java/org/codehaus/groovy/runtime/DefaultGroovyMethods.java

```diff
@@ -2273,6 +2273,26 @@ public static <T> T[] eachWithIndex(T[] self, @ClosureParams(value=FromString.cl
         return self;
     }
 
+    /**
+     * Iterates through an int[],
+     * passing each int and the element's index (a counter starting at
+     * zero) to the given closure.
+     * <pre class="groovyTestCase">
+     * int result = 0
+     * [0, 1, 2].eachWithIndex{ x, index {@code ->} result += x * index }
```

Comment:
Please ensure that the object expression is of type `int[]` to test the annotated method.  This is `List<Integer>` and so it tests another `eachWithIndex` veriant.  You can debug `MainJavadocAssertionTest` to see if your addition is hit by any test cases.

Reply:
I'm not sure how the debugging works, since I can't place the breakpoint inside the javadocs and I don't think I should step through all of the test cases.
I changed the test case to:
```
int[] nums = [5, 4, 3]
String result = ''
nums.eachWithIndex{ num, index -> result += "$index:${num + 1}" }
assert result == '0:61:52:4'
```

Reply:
To debug a junit test class, you can start gradle like this:
> gradlew :test --tests MainJavadocAssertionTest --no-daemon --debug-jvm

Gradle will start up and build enough to start the requested test(s) and wait for a connection on port 5005.  You can connect and debug with your favorite debugger / development environment.